### PR TITLE
doc: note how to install cert-manager

### PR DESCRIPTION
**Pull request by knight42**
Signed-off-by: Jian Zeng <anonymousknight96@gmail.com>

#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

We depends on cert-manager to handle webhook server certificates now, so we need to tell users to make user `cert-manager` has been installed before using `kueue`.

#### Which issue(s) this PR fixes:

Ref #232

### CODE REVIEW

File: docs/setup/install.md [hunk 1]
```diff
@@ -52,5 +53,5 @@ IMAGE_REGISTRY=registry.example.com/my-user make image-build image-push deploy
 To uninstall Kueue, run the following command:
 
 ```sh
-$ make undeploy 
-```
\ No newline at end of file
+$ make undeploy
```
Comment (alculquicondor) [inline docs/setup/install.md:56]: We are not putting `$` in the rest of the documentation.

```suggestion
make undeploy
```

File: docs/setup/install.md [hunk 2]
```diff
@@ -7,6 +7,7 @@ Make sure the following conditions are met:
 - A Kubernetes cluster with version 1.21 or newer is running. Learn how to [install the Kubernetes tools](https://kubernetes.io/docs/tasks/tools/).
 - The SuspendJob [feature gate](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/) is enabled. In Kubernetes 1.22 or newer, the feature gate is enabled by default.
 - The kubectl command-line tool has communication with your cluster.
+- The [cert-manager](https://github.com/cert-manager/cert-manager) has been installed in your Kubernetes cluster. Learn how to [install cert-manager](https://cert-manager.io/docs/installation/).
```
Comment (alculquicondor) [inline docs/setup/install.md:10]:
```suggestion
- The [cert-manager](https://github.com/cert-manager/cert-manager) is installed in your cluster. Learn how to [install cert-manager](https://cert-manager.io/docs/installation/).
```

**Review by alculquicondor — COMMENTED** [review]
/approve

Please squash

**Comment by knight42** [comment]
@alculquicondor Squashed 🚀

**Comment by alculquicondor** [comment]
/lgtm

Do you plan to work on the internal certificate issuing?

It's preferred that the installation of cert-manager is optional.

**Pull request merged**